Samples/WinMLSamplesGallery/TransformAsync/SegmentModel.h

```diff
@@ -0,0 +1,152 @@
+//#include <DXProgrammableCapture.h>
+#include "External/common.h"
+
+
+using namespace winrt::Microsoft::AI::MachineLearning;
+using namespace winrt::Microsoft::AI::MachineLearning::Experimental;
+using namespace winrt::Windows::Graphics::DirectX::Direct3D11;
+using namespace winrt::Windows::Media;
+
+// Model-agnostic helper LearningModels
+LearningModel Normalize0_1ThenZScore(long height, long width, long channels, const std::array<float, 3>& means, const std::array<float, 3>& stddev);
+LearningModel ReshapeFlatBufferToNCHW(long n, long c, long h, long w);
+LearningModel Invert(long n, long c, long h, long w);
+
+class StreamModelBase
+{
+public:
+	StreamModelBase(): 
+		m_inputVideoFrame(NULL),
+		m_outputVideoFrame(NULL),
+		m_session(NULL),
+		m_binding(NULL), 
+		m_syncStarted(FALSE)
+	{}
+	StreamModelBase(int w, int h) :
+		m_inputVideoFrame(NULL),
+		m_outputVideoFrame(NULL),
+		m_session(NULL),
+		m_binding(NULL),
+		m_syncStarted(FALSE)
+	{}
+	virtual ~StreamModelBase() {
+		if(m_session) m_session.Close();
+		if(m_binding) m_binding.Clear();
+		if (m_inputVideoFrame) m_inputVideoFrame.Close();
+		if (m_outputVideoFrame) m_outputVideoFrame.Close(); 
+	};
+
+	virtual void InitializeSession(int w, int h) =0;
+	virtual void Run(IDirect3DSurface src, IDirect3DSurface dest) =0;
+	
+	// Synchronous eval status
+	BOOL m_syncStarted; 
+	VideoFrame m_outputVideoFrame;
+	static const int m_scale = 5;
+
+protected:
+	// Cache input frames into a shareable d3d-backed VideoFrame
+	void SetVideoFrames(VideoFrame inVideoFrame, VideoFrame outVideoFrame) 
+	{
+		if (true || !m_videoFramesSet)
+		{
+			auto device = m_session.Device().Direct3D11Device();
+			auto inDesc = inVideoFrame.Direct3DSurface().Description();
+			auto outDesc = outVideoFrame.Direct3DSurface().Description();
+			/*
+				NOTE: VideoFrame::CreateAsDirect3D11SurfaceBacked takes arguments in (width, height) order
+				whereas every model created with LearningModelBuilder takes arguments in (height, width) order. 
+			*/ 
+			auto format = winrt::Windows::Graphics::DirectX::DirectXPixelFormat::B8G8R8X8UIntNormalized;
+			m_inputVideoFrame = VideoFrame::CreateAsDirect3D11SurfaceBacked(format, m_imageWidthInPixels, m_imageHeightInPixels, device);
+			m_outputVideoFrame = VideoFrame::CreateAsDirect3D11SurfaceBacked(format, m_imageWidthInPixels, m_imageHeightInPixels, device);
+			m_videoFramesSet = true;
+		}
+		// TODO: Fix bug in WinML so that the surfaces from capture engine are shareable, remove copy. 
+		inVideoFrame.CopyToAsync(m_inputVideoFrame).get();
+		outVideoFrame.CopyToAsync(m_outputVideoFrame).get();
+	}
+
+	void SetImageSize(int w, int h) {
+		m_imageWidthInPixels = w;
+		m_imageHeightInPixels = h;
+	}
+
+	LearningModelSession CreateLearningModelSession(const LearningModel& model, bool closedModel = true) {
+		auto device = m_useGPU ? LearningModelDevice(LearningModelDeviceKind::DirectXHighPerformance) : LearningModelDevice(LearningModelDeviceKind::Default);
```

Comment:
>m_useGPU ? LearningModelDevice(LearningModelDeviceKind::DirectXHighPerformance) : LearningModelDevice(LearningModelDeviceKind::Default); [](http://example.com/codeflow?start=16&length=136)

Propose the sligthly shorter:

`= LearningModelDevice(m_useGPU ?  LearningModelDeviceKind::DirectXHighPerformance : LearningModelDeviceKind::Default);`